controllers/smbshare_controller.go

```diff
@@ -62,7 +62,8 @@ func (r *SmbShareReconciler) Reconcile(
 	reqLogger.Info("Reconciling SmbShare")
 
 	smbShareManager := resources.NewSmbShareManager(
-		r, r.Scheme(), r.recorder, reqLogger)
+		r, r.Scheme(), r.recorder, reqLogger) // nolint:typecheck
```

Comment:
Just for my education: what kind of linter error was this generating without the suppresion comment?

Reply:
```r.Scheme undefined (type *SmbShareReconciler has no field or method Scheme)```

Reply:
That's weird. Probably a bug in the linter - or else this code shouldn't compile :-)

Reply:
Indeed -- probably linter bug